### PR TITLE
Replace util_rand with built-in rand

### LINKED ISSUE #6115: Random title screen music not random on launch
**OS:** Arch Linux
**Version:** 0.1.0
**Commit/Build:** 34614b2

I have selected random title screen music in settings and have installed both rct2 and rct1 correctly. I noticed when casually playing the game that I'm hearing the rct2 title screen music much more often than the rct1 one, so I tested to see if this was true. I launched the game a few dozen times and all of the times the rct2 music started playing. When coming back from a saved game the title screen music does seem random as is the case when you change the option in settings. So it looks like the music isn't random only just after launching the game.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

**Steps to reproduce:**
1. Setup rct1 folder
2. Set title screen music to random
3. Launch the game
4. Listen to not so random rct2 music

**Dump file**

**Screenshots / Video:**

**Save game:**

### PR DESCRIPTION
**Pull request by flipflop97**
The util_rand (and util_srand) function were unnecessary because C has its own rand function, util_rand was also causing problems (#6115), so I replaced all calls to util_rand and util_srand with rand and srand. This seems to have indeed fixed #6115. I also removed the util_ functions from the code.

### CODE REVIEW

**Comment by IntelOrca** [comment]
It was originally `rand` before. I can't remember why it was changed though... @duncanspumpkin ?

**Comment by duncanspumpkin** [comment]
Yes there was a reason behind its existence but i can't remember right now

**Comment by ZehMatt** [comment]
Wouldn't it be a lot easier to place rand() in util_rand so you don't need to actually replace all references?

**Comment by pizza2004** [comment]
I'm not positive on this but I think it may have been replaced because different OS will implement Rand differently sometimes which could cause network disconnects.

**Comment by IntelOrca** [comment]
`util_rand` is not used for anything that needs synchronising across network. It is not required to be the same across all platforms.

**Comment by Gymnasiast** [comment]
https://github.com/OpenRCT2/OpenRCT2/commit/5bd3b8d6afa9b3d27bf499a52bd5da37906615b8

^ According to the commit message, rand() would have a poor range on some platforms. It doesn't say what those platforms are.

**Comment by IntelOrca** [comment]
In which case we should still use our wrapper as @ZehMatt said.
@flipflop97 could you instead just change the implementation of `util_rand` to call `rand` instead?

**Comment by flipflop97** [comment]
Done, I left the original code commented out in case of rand not working properly on some platforms.

File: src/openrct2/util/util.c [hunk 1]
```diff
@@ -400,24 +400,32 @@ bool str_is_null_or_empty(const char *str)
     return str == NULL || str[0] == 0;
 }
 
-uint32 srand0, srand1, srand2, srand3;
+//uint32 srand0, srand1, srand2, srand3;
 
 void util_srand(sint32 source) {
+    srand(source);
+    
+    /* Old implementation, use for specific platforms when srand() causes problems.
```
Comment (AaronVanGeffen) [inline src/openrct2/util/util.c]: I think you can just remove this rather than commenting it out altogether. That's what version control is for.

Comment (flipflop97) [inline src/openrct2/util/util.c]: Yes you're probably right, I just thought if it's causing problems again soon it could be added back easily. I'll remove it.

**Comment by duncanspumpkin** [comment]
Okay I remember now why we replaced it with util rand. It was to do with `crashed_vehicle_particle_create` it was expecting a value between `0xFFFF` and `0x0000` but VS's `rand` had a max of `0x7FFF` as defined as the minimum by the C standard. This meant that crashed particle sprites would always have a positive acceleration. 

Since then we replaced `util_rand` with `scenario_rand` in `crashed_vehicle_particle_create` as part of network desync fix so we don't actually need it to go to `0xFFFF`. 

So to summarise this shouldn't have an issue but remember that this will only produce numbers up to `0x7FFF` on some platforms.

**Comment by Gymnasiast** [comment]
@duncanspumpkin I prefer an actual comment on the function. We forgot it last time, we will probably forget it again.
